apps/btc_family/btc_pub_key.c

```diff
@@ -204,7 +204,17 @@ static size_t btc_get_address(const uint8_t *seed,
     case NON_SEGWIT:
       hdnode_get_address(&node, g_btc_app->p2pkh_addr_ver, addr, 35);
       break;
-    // TODO: add support for taproot and segwit
+    case PURPOSE_SEGWIT:
+      ecdsa_get_address_segwit_p2sh(
+              node.public_key,
```

Comment:
just take &node as input

Reply:
This function is part of the crypto library api. I can't really modify it